### DISCUSSION TITLE
uneeded buffers spawned

### DISCUSSION THREAD

**Discussion by rubiin** [olimorris/persisted.nvim#133] [General]
![image](https://github.com/olimorris/persisted.nvim/assets/8222059/8c7bcd75-5935-4f37-9e10-0d4de505036f)

These two buffers(No name and folder directory)  spawn whenever i restore last session
![image](https://github.com/olimorris/persisted.nvim/assets/8222059/852d9a14-1b97-45a8-be8b-cfb39d5483ee)

**Comment by olimorris** [comment]
This will be because they're written into the session file. The plugin is not creating them. I'd advise using [callbacks](https://github.com/olimorris/persisted.nvim?tab=readme-ov-file#events--callbacks) to handle this.

**Reply by rubiin** [reply]
@olimorris  you mean this
```
      should_autosave = function()
        local excluded_filetypes = {
          "alpha",
          "oil",
          "lazy",
          "spectre_panel",
          "",
        }

        for _, filetype in ipairs(excluded_filetypes) do
          if vim.bo.filetype == filetype then
            return false
          end
        end

        return true
      end,
    },

```

**Reply by olimorris** [reply]
No, the callbacks - https://github.com/olimorris/persisted.nvim?tab=readme-ov-file#events--callbacks.

The should_autosave will restrict saving of a session when you're in a buffer type. The Callbacks are functions that can execute at a given point in time. So you could call a function which closes a plugin like NvimTree (which often leaves empty buffers in session files)
